### PR TITLE
Update bingads SOAP methods.

### PR DESCRIPTION
**Pull request by przywartya**
# Description of change
(write a short description or paste a link to JIRA)

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_bing_ads/fetch_ad_accounts.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    customer_id = None
+
+    customer_id = response["s:Envelope"]["s:Body"]["GetCustomersInfoResponse"][
+        "CustomersInfo"
+    ]["a:CustomerInfo"]["a:Id"]
+
+    assert customer_id and isinstance(customer_id, str)
+
+    try:
```
Comment (JingLin0) [inline tap_bing_ads/fetch_ad_accounts.py]: It's better you make 
```
try:
except:
```
as one function.

**Review by JingLin0 — APPROVED** [review]

File: tap_bing_ads/fetch_ad_accounts.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                authentication_token=access_token, developer_token=developer_token,
+            ),
+        )
+        response.raise_for_status()
+    except HTTPError as http_err:
+        logger.error(f"HTTP error occurred: {http_err}")
+    except Exception as err:
+        logger.error(f"Other error occurred: {err}")
+
+    response = response.content.decode("utf-8")
```
Comment (ghost) [inline tap_bing_ads/fetch_ad_accounts.py]: what if it fails to decode, or parse or access response? 
Everything has to be in the try/except

**Review by ghost — CHANGES_REQUESTED** [review]
cover all soap parsing and requests in try/except

File: tap_bing_ads/fetch_ad_accounts.py (hunk 2 above)
Comment (przywartya) [reply tap_bing_ads/fetch_ad_accounts.py]: Ok I changed it, please review

File: tap_bing_ads/fetch_ad_accounts.py (hunk 1 above)
Comment (przywartya) [reply tap_bing_ads/fetch_ad_accounts.py]: Ok I changed it, please review

**Review by JingLin0 — APPROVED** [review]

File: tap_bing_ads/fetch_ad_accounts.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+        return response
+    except HTTPError as http_err:
+        logger.error(f"HTTP error occurred: {http_err}")
+    except UnicodeError as unicode_err:
+        logger.error(f"Unicode decoding error occurred: {unicode_err}")
+    except Exception as err:
+        logger.error(f"Other error occurred: {err}")
+
+    raise Exception("Could not process request response")
```
Comment (paddie) [inline tap_bing_ads/fetch_ad_accounts.py]: remove all the try-catches in favour of something like this: https://github.com/singer-io/singer-python/blob/master/singer/utils.py#L222

**Review by paddie — CHANGES_REQUESTED** [review]

File: tap_bing_ads/fetch_ad_accounts.py (hunk 2 above)
Comment (przywartya) [reply tap_bing_ads/fetch_ad_accounts.py]: So now it is handled by a top-level error handler and we don't have to worry about it 

https://github.com/dreamdata-io/tap-bing-ads/pull/3#discussion_r389722360

**Review by JingLin0 — APPROVED** [review]

**Review by paddie — APPROVED** [review]
Looks good, sorry for the delay!

**Pull request merged**